### DISCUSSION TITLE
Manifest verification

### DISCUSSION THREAD

**Discussion by Jubilee101** [pgmoneta/pgmoneta#143] [Ideas]
Hi @jesperpedersen , I'm starting to work on manifest checksum verification. The manifest is in a format very much like json, but I couldn't find a good json parser library in c that can be easily installed. So I think I'll just parse the manifest manually. Let me know if you have better idea.

**Comment by Jubilee101** [comment]
Also, I'm not sure where to put the functionalities. Right now I just add them under `utils.c`. It might be worth it to create a `manifest.c`, which handles the parsing and verification.

**Reply by Jubilee101** [reply]
Oh I think we can use cJson perhaps: https://github.com/DaveGamble/cJSON

**Reply by jesperpedersen** [reply]
Yeah, that one should be ok

**Comment by jesperpedersen** [comment]
Having a simple solution - within the code base - is also ok

**Reply by Jubilee101** [reply]
cJSON seems cute and actively maintained. I'll try if it works and parse it manually if it doesn't.

**Comment by jesperpedersen** [comment]
Doing `manifest.h` and `manifest.c` with the functionality would be good.

I expect that manifest functionality over time will expand since the server is doing it anyway
